app/templates/components/goodcity/item-search-overlay.hbs

```diff
@@ -26,7 +26,7 @@
             {{#if displayResults}}
               {{#infinite-list height="85vh" loadMore=(action "loadMoreItems") as |items| }}
                 {{#each items as |item|}}
-                  <div class="row" {{action 'selectItem' item}}>
+                  <div class="row" {{action 'selectItem' item cb}}>
```

Comment:
if your goal with the callback is to have multiple actions trigger one after the other, you can check out the `multi-actions` helper I created in stock

works like this

```handlebars
 <button {{action (multi-actions
    (action "doThis")
    (action "doThat")
    (action "andThenThis")
  )
 }}>
```

Reply:
note: actions need to return a promise for it work work well. So `async` is encouraged

Reply:
FYI: completely optional, mentioning it just in case

Reply:
adding callback here was not necessary. I have updated it